File: 7.hs
{-
 - By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see
 - that the 6th prime is 13.
 - What is the 10001st prime number?
 -}

nthPrime :: Integer -> Integer
nthPrime n = last $ take (fromIntegral n) primes

primes :: [Integer]
primes = 2: 3: sieve (tail primes) [5,7..] where
    sieve (p:ps) xs = h ++ sieve ps [x | x <- t, rem x p /= 0] where
        (h,~(_:t)) = span(< p*p) xs

File: ruby/7.rb
# By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see
# that the 6th prime is 13.
# What is the 10001st prime number?

require "prime"

def get_prime(nth)
  Prime.first(nth).last
end

puts get_prime(10001)
